File: Data Structures/php/arraymanipulation.php
<?php

// Complete the arrayManipulation function below.
function arrayManipulation($n, $queries) {
    $arr = [];
    foreach($queries as $q){
        $arr[$q[0]] += $q[2];
        $arr[$q[1]+1] -= $q[2];
    }
    ksort($arr);
    
    $max = 0;
    $tmp = 0;
    foreach($arr as $a){
        $tmp += $a;
        if($tmp > $max){
            $max = $tmp;
        }
    }
    return $max;

}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$stdin = fopen("php://stdin", "r");

fscanf($stdin, "%[^\n]", $nm_temp);
$nm = explode(' ', $nm_temp);

$n = intval($nm[0]);

$m = intval($nm[1]);

$queries == array();

for ($i = 0; $i < $m; $i++) {
    fscanf($stdin, "%[^\n]", $queries_temp);
    $queries[] = array_map('intval', preg_split('/ /', $queries_temp, -1, PREG_SPLIT_NO_EMPTY));
}

$result = arrayManipulation($n, $queries);

fwrite($fptr, $result . "\n");

fclose($stdin);
fclose($fptr);

File: Data Structures/ArrayManipulation.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the arrayManipulation function below.
def arrayManipulation(n, queries)
    diffArr =[]
    for i in 0..n
        diffArr.push(0)
    end
    for j in queries
        diffArr[j[0]-1] += j[2]
        diffArr[j[1]] -= j[2]
    end
    max = 0
    sum = 0
    for k in diffArr
        sum += k
        max = [sum, max].max
    end
    return max

end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

nm = gets.rstrip.split

n = nm[0].to_i

m = nm[1].to_i

queries = Array.new(m)

m.times do |i|
    queries[i] = gets.rstrip.split(' ').map(&:to_i)
end

result = arrayManipulation n, queries

fptr.write result
fptr.write "\n"

fptr.close()
